src/docs/development/tools/devtools/code-size.md

```diff
@@ -0,0 +1,230 @@
+---
+title: Using the code size tool
+description: Learn how to use the DevTools code size tool.
+---
+
+## What is it?
+
+The code size tool allows you to analyze the total size of your app, including
+both Native and Dart code, or just the size of the Dart code in your app. You
+can view a single snapshot of "size information" using the [Analysis tab][], or
+compare two different snapshots of "size information" using the [Diff tab][].
+
+### What is "size information"?
+
+The Dart AOT compiler performs tree-shaking on your code when compiling your
+application. This means that the compiler attempts to optimize your app's size
+by removing pieces of code that are unused or unreachable.
```

Comment:
I like this paragraph, too! But I do have a question that I couldn't figure out by googling. Does the compiler perform tree shaking in debug, profile, AND release modes?

Reply:
for sure in release mode, not in debug mode (debug mode is JIT compiled), and I'm not sure on profile mode..? cc @mraleph

Reply:
treeshaking only performed in release and profile. not in debug.

Reply:
I only asked because I thought it could use clarification. But I'm not going to let that hold this PR up.